### PR TITLE
fixes #2587 - migration to increase varchar length of media.name and other string fields

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by domcleal** [comment]
@isratrade this doesn't seem to remove the limit for me on SQLite (at least):

<pre>
$ sqlite3 db/development.sqlite3
sqlite> .schema media
CREATE TABLE "media" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar(50) DEFAULT '' NOT NULL, "path" varchar(255) DEFAULT '' NOT NULL, "created_at" datetime, "updated_at" datetime, "media_path" varchar(128), "config_path" varchar(128), "image_path" varchar(128), "os_family" varchar(255));
$ rake db:migrate
==  ChangeColumnLengths: migrating ============================================
-- change_column(:media, :name, :string, {:null=>false})
   -> 0.1193s
etc etc...
$ sqlite3 db/development.sqlite3
sqlite> .schema media
CREATE TABLE "media" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar(50) DEFAULT '' NOT NULL, "path" varchar(255) DEFAULT '' NOT NULL, "created_at" datetime, "updated_at" datetime, "media_path" varchar(128), "config_path" varchar(128), "image_path" varchar(128), "os_family" varchar(255));
</pre>

The limit is still 50 chars?

File: db/migrate/20130530061844_change_column_lengths.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    change_column :models, :hardware_model, :string
+    change_column :operatingsystems, :name, :string
+    change_column :operatingsystems, :release_name, :string
+    change_column :operatingsystems, :type, :string
+    change_column :ptables, :name, :string, :null => false
+    change_column :roles, :name, :string
+  end
+
+  def down
+    change_column :media, :name, :string, :limit => 64, :null => false
```
Comment (domcleal) [inline db/migrate/20130530061844_change_column_lengths.rb]: I just rolled back and this is different to before, it used to be 50.  Might be worth double checking the others?

Comment (isratrade) [inline db/migrate/20130530061844_change_column_lengths.rb]: @domcleal, I checked manually on postgres and it migrates and rollbacks fine. ??

Comment (domcleal) [inline db/migrate/20130530061844_change_column_lengths.rb]: @isratrade it does rollback, but it was 50 chars to begin with and is now 64 chars

**Comment by isratrade** [comment]
@domcleal, re-commited and added :default => '' where necessary.  Somehow, these were missing on operatingsystems.major and .minor fields which was the culprit in my previous test failures.  Quoting your meme -- "I don't know how" :-)

rake db:migrate:up VERSION=20130530061844
rake db:migrate:down VERSION=20130530061844

**Comment by domcleal** [comment]
@isratrade I'm seeing the same issue on SQLite, the limit doesn't get removed... I don't think that has anything to do with the default value?

<pre>
sqlite> .schema media
CREATE TABLE "media" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar(50) DEFAULT '' NOT NULL, "path" varchar(255) DEFAULT '' NOT NULL, "created_at" datetime, "updated_at" datetime, "media_path" varchar(128), "config_path" varchar(128), "image_path" varchar(128), "os_family" varchar(255));
</pre>

**Comment by isratrade** [comment]
@domcleal, I explicitly added the :limit => 255.  the sqlite adapter is probably correct is not updating the varchar length unless it's pass.  The pg adapter apparently assumes 255 unless told otherwise.

File: db/migrate/20130530061844_change_column_lengths.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    change_column :auth_sources, :name, :string, :null => false, :default => '', :limit => 255
+    change_column :auth_sources, :host, :string, :limit => 255
+    change_column :auth_sources, :account_password, :string, :limit => 255
+    change_column :auth_sources, :attr_login, :string, :limit => 255
+    change_column :auth_sources, :attr_firstname, :string, :limit => 255
+    change_column :auth_sources, :attr_lastname, :string, :limit => 255
+    change_column :auth_sources, :attr_mail, :string, :limit => 255
+    change_column :domains, :fullname, :string, :limit => 255
+    change_column :features, :name, :string, :limit => 255
+    change_column :hosts, :mac, :string, :default => '', :limit => 255
```
Comment (domcleal) [inline db/migrate/20130530061844_change_column_lengths.rb:18]: Unnecessary?

**Comment by domcleal** [comment]
Thanks @isratrade, this looks good for me now on SQLite.  Waiting for tests now.

**Comment by domcleal** [comment]
Merged as ea5a012, thanks @isratrade!

**Pull request closed without merging**